reference/api/settings.md

```diff
@@ -1,201 +1,1391 @@
+| **`index_uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |
+
+#### Example
+
+<CodeSamples id="get_pagination_settings_1" />
+
+##### Response: `200 OK`
+
+```json
+{
+  "maxTotalHits": 1000
+}
+```
+
+### Update pagination settings
+
+<RouteHighlighter method="PATCH" route="/indexes/{index_uid}/settings/pagination"/>
+
+Partially update the pagination settings for an index.
+
+#### Path parameters
+
+| Name              | Type   | Description                                                               |
+| :---------------- | :----- | :------------------------------------------------------------------------ |
+| **`index_uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |
+
+#### Body
+
+| Name               | Type    | Default value | Description                                          |
+| :----------------- | :------ | :------------ | :--------------------------------------------------- |
+| **`maxTotalHits`** | Integer | `1000`        | The maximum number of results Meilisearch can return |
+
+`maxTotalHits` takes priority over search parameters such as `limit` and `offset`.
+
+For example, if you set `maxTotalHits` to 100, you will not be able to access search results beyond 100 no matter the value configured for `offset`.
+
+::: note
+Setting `maxTotalHits` to a high value might negatively impact performance and expose instance data to malicious scraping.
+:::
+
+#### Example
+
+<CodeSamples id="update_pagination_settings_1" />
+
+##### Response: `202 Accepted`
+
+```json
+{
+  "taskUid": 1,
+  "indexUid": "books",
+  "status": "enqueued",
+  "type": "settingsUpdate",
+  "enqueuedAt": "2022-04-14T20:56:44.991039Z"
+}
+```
+
+You can use the returned `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).
+
+### Reset pagination settings
+
+Reset an index's pagination settings to their [default value](#pagination-object).
+
+#### Path parameters
+
+| Name              | Type   | Description                                                               |
+| :---------------- | :----- | :------------------------------------------------------------------------ |
+| **`index_uid`** * | String | [`uid`](/learn/core_concepts/indexes.md#index-uid) of the requested index |
+
+#### Example
+
+<CodeSamples id="reset_pagination_settings_1" />
+
+##### Response: `202 Accepted`
+
+```json
+{
+  "taskUid": 1,
+  "indexUid": "books",
+  "status": "enqueued",
+  "type": "settingsUpdate",
+  "enqueuedAt": "2022-04-14T20:53:32.863107Z"
+}
+```
+
+You can use the returned `taskUid` to get more details on [the status of the task](/reference/api/tasks.md#get-one-task).
+
+## Ranking rules
+
+Ranking rules are built-in rules that allow you to customize the relevancy of your search results. They are stored in an array and applied in order of appearance.
```

Comment:
I found "order of appearance" potentially confusing so this is my attempt to improve it. Feel free to veto if you disagree

```suggestion
Ranking rules are built-in rules that rank search results according to certain criteria. They are applied in the same order in which they appear in the `rankingRules` array.
```